components/layout_2020/flow/inline/mod.rs

```diff
@@ -2039,58 +2028,85 @@ impl IndependentFormattingContext {
                     },
                 };
 
-                let content_rect = PhysicalRect::new(
-                    PhysicalPoint::zero(),
-                    LogicalVec2 {
-                        block: block_size,
-                        inline: inline_size,
-                    }
-                    .to_physical_size(container_writing_mode),
-                );
+                let content_size = LogicalVec2 {
+                    block: block_size,
+                    inline: inline_size,
+                }
+                .to_physical_size(container_writing_mode);
+                let content_rect = PhysicalRect::new(PhysicalPoint::zero(), content_size);
 
                 (
                     independent_layout.fragments,
                     content_rect,
                     Some(independent_layout.baselines),
-                    Some(positioning_context),
                 )
             },
         };
 
-        // Offset the content rectangle by the physical offset of the padding, border, and margin.
-        let pbm_physical_offset = pbm_sums
-            .start_offset()
-            .to_physical_size(container_writing_mode);
-        let content_rect = content_rect.translate(pbm_physical_offset.to_vector());
-
-        let fragment = BoxFragment::new(
+        let mut fragment = BoxFragment::new(
             self.base_fragment_info(),
             self.style().clone(),
             fragments,
             content_rect,
             pbm.padding.to_physical(container_writing_mode),
             pbm.border.to_physical(container_writing_mode),
             margin.to_physical(container_writing_mode),
+            // Floats can have clearance, but it's handled internally by the float placement logic,
+            // so there's no need to store it explicitly in the fragment.
+            // And atomic inlines don't have clearance.
             None, /* clearance */
             CollapsedBlockMargins::zero(),
         );
 
+        if is_float {
+            return fragment;
+        }
+
+        // Offset the content rectangle by the physical offset of the padding, border, and margin.
+        let pbm_physical_offset = pbm_sums
+            .start_offset()
+            .to_physical_size(container_writing_mode);
+        fragment.content_rect = fragment
+            .content_rect
+            .translate(pbm_physical_offset.to_vector());
```

Comment:
This feels like it belongs a bit closer to inline layout, could you return `pbm_sums` from this function and have it take place in inline layout?